### PR TITLE
Added tests that check for null exceptions

### PR DESCRIPTION
**Pull request by AndraVasilache**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-

## Checklist: I completed these to help reviewers :)

- [ ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by gitcliff** [comment]
@AndraVasilache the link to the ticket is wrong

**Comment by AndraVasilache** [comment]
https://issues.openmrs.org/browse/TRUNK-5666

**Comment by Andu033** [comment]
The code looks good to me. And the coverage increased.

**Comment by dkayiwa** [comment]
Can you include the ticket id in the commit message as recommended at https://wiki.openmrs.org/display/docs/Pull+Request+Tips

File: api/src/test/java/org/openmrs/util/UserByNameComparatorTest.java [hunk 1]
```diff
@@ -63,4 +63,63 @@ public void compare_shouldSortUsersByPersonNames() {
...
+		listToSort.add(user3);
+		listToSort.add(user2);
+		listToSort.add(user1);
+		listToSort.add(user4);
+
+		listToSort.sort(new UserByNameComparator());
+
+		Iterator<User> it = listToSort.iterator();
+		Assert.assertTrue("Expected user1 to be the first in the sorted user list but wasn't", user1.equals(it.next()));
+		Assert.assertTrue("Expected user3 to be the second in the sorted user list but wasn't", user3.equals(it.next()));
```
Comment (khush3211) [inline api/src/test/java/org/openmrs/util/UserByNameComparatorTest.java]: Can we import Assert once above and simply use `assertTrue(...)`?

**Pull request closed without merging**